lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3469,7 +3491,9 @@ void AttributeChecker::visitDifferentiatingAttr(DifferentiatingAttr *attr) {
     // indices. Reject duplicate attributes for the same decl and parameter
     // indices pair.
     if (!insertion.second && insertion.first->getSecond() != da) {
-      diagnoseAndRemoveAttr(da, diag::differentiable_attr_duplicate);
+      TC.diagnose(da->getLocation(), diag::differentiable_attr_duplicate);
+      TC.diagnose(insertion.first->getSecond()->getLocation(),
+                  diag::differentiable_attr_duplicate_note);
```

Comment:
Do you not need to remove it?

Reply:
I thought removing was incorrect for some reason, but removing seems correct. Reverted.